### ISSUE TITLE
Remove WH_ARCH_YAML from models and ops

### ISSUE THREAD

**Issue by aliuTT** [tenstorrent/tt-metal#11059]
WH_ARCH_YAML is removed from runtime backend, will need model writers to update tests to use the new APIs:

**Comment by mtairum** [comment]
Will this be back-propagated to all instances where the flag is used?

Or is it up to each model owner to check their model and update it?

**Comment by aliuTT** [comment]
I updated it for all the Metal CPP tests and a few fixtures, but I saw a lot of model/op specific tests that had this flag. I didn't feel confident modifying all of them, so leaving it to model owner to check their own tests.

**Comment by aliuTT** [comment]
Ping on this, any plans to get `WH_ARCH_YAML` removed from tests?

**Comment by mtairum** [comment]
@aliuTTYes, but low priority at the moment.

**Comment by ayerofieiev-tt** [comment]
How many entires are there in total?

**Comment by aliuTT** [comment]
Ping on this issue, we have customer team still broadcasting `WH_ARCH_YAML` in documentation and new models.

**Comment by blozano-tt** [comment]
Team, any status update? We have identified this issue as a customer impacting quality of life item.

**Comment by skhorasganiTT** [comment]
I don't think the team has had time to make progress on this due to other higher priority tasks, we may want to raise the priority level of this if it needs to be done soon.

**Comment by uaydonat** [comment]
@handrewsTT can you take a look?

**Comment by uaydonat** [comment]
Comment from @aliuTT 

> There's two issues here:
> detect n300 (which we have an API for in runtime ClusterType Cluster::get_cluster_type() but maybe not piped to ttnn layer)
> even after detecting n300, the proper way to set ethernet dispatch is to just pass in dispatch_core_type = ttnn.device.DispatchCoreType.ETH to the test fixture.  The current env var passing was to enable model tests that were never cleaned up: https://github.com/tenstorrent/tt-metal/issues/11059

**Comment by skhorasganiTT** [comment]
Is there any reason why eth dispatch shouldn't be enabled by default on wh devices that don't have 8x8 core grid (and instead make it an opt-out option)? Since we already use it on most of our models (speaking more on behalf of the LLMs, might need some input on the CNNs / vision models), we could then just delete that env var from the readmes and not modify tests which use it. cc @ayerofieiev-tt

**Comment by blozano-tt** [comment]
> Is there any reason why eth dispatch shouldn't be enabled by default on wh devices that don't have 8x8 core grid (and instead make it an opt-out option)? Since we already use it on all our models, we could then just delete that env var from the readmes and not modify tests. cc [@ayerofieiev-tt](https://github.com/ayerofieiev-tt)

@pgkeller @jbaumanTT are the experts here.

**Comment by pgkeller** [comment]
That works for me.  @nhuang-tt can you look into this?

**Comment by skhorasganiTT** [comment]
@uaydonat @tt-rkim do you know whether we use eth dispatch for CNNs / vision models? (In contrast to the LLMs which use it everywhere)

**Comment by tt-rkim** [comment]
@pavlejosipovic and @s-jovic can answer more, but I believe mostly yes they do use eth dispatch, with some small exceptions

<img width="564" height="501" alt="Image" src="https://github.com/user-attachments/assets/060c49a6-d42c-4cbf-9b78-4d0e1855c8f4" />

**Comment by pavlejosipovic** [comment]
I would make it a default on n300

**Comment by tt-rkim** [comment]
There are probably quite a few unit tests that don't work with eth dispatch on ttnn ops...

Does someone have to go through those one by one and turn it off if it is made default?

**Comment by pavlejosipovic** [comment]
I don't think so as our ci like post commit and l2 nightly run on both n300 and n150 so we had to deal with making sure tests work on both grids or skip them if they are tailored for just one of them.

**Comment by TT-BrianLiu** [comment]
You can also consider using a device fixture with a smaller scope (ie. adding a `conftest.py` in models folder). Might be better to not overload one global device fixture with small things like this

**Comment by handrewsTT** [comment]
Okay, first things first we need to be able to detect N300 programmatically per the discussions on slack. There is an open PR https://github.com/tenstorrent/tt-metal/pull/25493 to add this functionality to ttnn. Running APC now but it's ready for eyes whenever

**Comment by handrewsTT** [comment]
Once we're happy with this PR we can then work on updating all the tests and the default behaviour to remove all dependency on the `WH_ARCH_YAML` env variable

**Comment by nhuang-tt** [comment]
I am running CI pipelines on a branch with `WH_ARCH_YAML` removed now.

Just to confirm, we are only setting Ethernet dispatch by default on N300/T3K for the Pytests. This will not be a tt-metal default.

**Comment by handrewsTT** [comment]
@nhuang-tt the MR above only exposes the ability to detect the cluster type through ttnn. We still need to add the logic to handle the dispatch core config differently for N300/T3K (I'm working on this). Unless your branch handles this?

**Comment by handrewsTT** [comment]
There is a desire to change the default behaviour in the `CreateDevice` API itself so that users don't have to specify ETH dispatch on N300/T3K and it is just set by default. We can handle this in the TTNN API using the newly exposed get_cluster_type functionality but it might break some unit tests that require worker dispatch on N300. There was a preference to fix these tests rather than falling back on manually setting the dispatch config for every test where we want ETH dispatch on N300.

**Comment by nhuang-tt** [comment]
I removed all the references in `nhuang/remove-ttnn-wh-arch-yaml`. Most of the work is search and replace so you can do you work based off this branch. In conftest I used your new API to check if it's N300/T3K and set the core type to Ethernet for the test Device but several CI runs are failing --- I am guessing due to some tests not supporting Eth dispatch.

Note, for the Profiler test, they have a test for Dispatch on Worker cores so I added an environment var in get_dispatch_core_type for the profiler test to force worker dispatch.
